addons/sourcemod/scripting/Leader.sp

```diff
@@ -620,56 +826,338 @@ public Action RemoveTheLeader(int client, int args)
 // Purpose:
 //----------------------------------------------------------------------------------------------------
 public Action Leader(int client, int args)
-{
-	if(CheckCommandAccess(client, "sm_admin", ADMFLAG_GENERIC, false))
-	{
-		if(args == 1)
-		{
-			char arg1[65];
-			GetCmdArg(1, arg1, sizeof(arg1));
-			int target = FindTarget(client, arg1, false, false);
-			if (target == -1)
+{	
+	char arg1[65];
+	GetCmdArg(1, arg1, sizeof(arg1));
+
+	if(g_iClientNextLeader[client] < GetTime())
+	{			
+		if(args >= 1)
+		{	
+			if(CheckCommandAccess(client, "sm_admin", ADMFLAG_GENERIC, false))
 			{
-				return Plugin_Handled;
-			}
+				int target = FindTarget(client, arg1, false, false);
+				if (target == -1)
+				{
+					return Plugin_Handled;
+				}
 
-			if(target == leaderClient)
+				if(g_bIsClientALeader[target])
+				{
+					LeaderMenu(target);
+					return Plugin_Handled;
+				}
+				else
+				{
+					if(GetCurrentLeadersCount() != leaders_max)
+					{
+						if(IsPlayerAlive(target) && ZR_IsClientHuman(target))
+						// Admin Target a player
+						{
+							SetLeader(target, -1);
+							CPrintToChatAll("{green}[SM] {olive}%N {default}is now a new leader !", target);
+							CPrintToChat(target, "{green}[SM] {default}Type {green}!leader {default}to open up the leader menu.");
+							LeaderMenu(target);
+							LogAction(client, target, "[Leader] \"%L\" is now a new leader ! (Designed by \"%L\")", target, client);
+							return Plugin_Handled;
+						}
+						else
+						{
+							CReplyToCommand(client, "{green}[SM] {default}The target has to be an alive Human !");
+							return Plugin_Handled;
+						}
+					}
+					else if(GetCurrentLeadersCount() == leaders_max && leaders_max != 1)
+					{
+						CReplyToCommand(client, "{green}[SM] {default}Oops, there are %d leaders which is the max number of available leaders on this map.", leaders_max);
+						CReplyToCommand(client, "{green}[SM] {default}Type {olive}sm_replaceleader <a current leader> <target>{default} to replace a current leader with a target");
```

Comment:
CReplyToCommand(client, "{green}[SM] {default}Type {olive}sm_replaceleader{default} to replace an active leader with %N", target)